src/main/java/com/adyen/httpclient/AdyenHttpClient.java

```diff
@@ -181,35 +183,63 @@ private URI createUri(String endpoint, Map<String, String> params) throws HTTPCl
     }
 
     private CloseableHttpClient createCloseableHttpClient(Config config) throws HTTPClientException {
-        CloseableHttpClient httpclient;
 
-        if (config.getTerminalCertificate() != null) {
-            HttpClientBuilder httpClientBuilder = HttpClients.custom();
+        if (config.getClientKeyStore() != null && config.getTrustKeyStore() != null) {
+            return HttpClients.custom()
+                    .setSSLSocketFactory(getClientCertificateAuthSSLContext(config))
+                    .build();
+        } else if (config.getTerminalCertificate() != null) {
+            return HttpClients.custom()
+                    .setSSLSocketFactory(getTerminalCertificateSocketFactory(config))
+                    .build();
+        } else {
+            return HttpClients.createDefault();
+        }
+    }
+
+    private SSLConnectionSocketFactory getTerminalCertificateSocketFactory(Config config) throws HTTPClientException {
+        try {
             // Create new KeyStore for the terminal certificate
-            try {
-                KeyStore keyStore = KeyStore.getInstance(KeyStore.getDefaultType());
-                keyStore.load(null, null);
-                keyStore.setCertificateEntry(TERMINAL_CERTIFICATE_ALIAS, config.getTerminalCertificate());
-
-                TrustManagerFactory trustFactory =
-                        TrustManagerFactory.getInstance(TrustManagerFactory.getDefaultAlgorithm());
-                trustFactory.init(keyStore);
-                TrustManager[] trustManagers = trustFactory.getTrustManagers();
-
-                // Install the terminal certificate trust manager
-                SSLContext sc = SSLContext.getInstance(SSL);
-                sc.init(null, trustManagers, new java.security.SecureRandom());
-
-                SSLConnectionSocketFactory sslConSocFactory = new SSLConnectionSocketFactory(sc, createHostnameVerifier(config.getEnvironment()));
-                httpClientBuilder.setSSLSocketFactory(sslConSocFactory);
-            } catch (GeneralSecurityException | IOException e) {
-                throw new HTTPClientException("Error loading certificate from path", e);
+            KeyStore keyStore = KeyStore.getInstance(JAVA_KEYSTORE);
```

Comment:
should we use `KeyStore.getDefaultType` like suggested in #567?
```suggestion
            KeyStore keyStore = KeyStore.getInstance(KeyStore.getDefaultType());
```

Reply:
> should we use `KeyStore.getDefaultType` like suggested in #567?

corrected it, thanks